examples/render-md-server-side/app/routes/index.tsx

```diff
@@ -0,0 +1,40 @@
+import { useLoaderData } from "remix";
+import MarkdownContainer from "../markdown";
+
+/**
+ * Mock function to fetch your markdown content.
+ * You could fetch from any CMS, the filesystem, GitHub, etc.
+ */
+async function fetchMarkdownFormSomewhere() {
+  return `
+# This is a H1 tag using the custom H1 component
+
+## This is a H2 tag
+
+### This is a H3 tag
+
+This is a paragraph
+
+This is a paragraph with a [link](https://remix.run)
+
+This is a paragraph with a \`code\` block
+
+\`\`\`javascript
+console.log("Remix rocks!");
+\`\`\`
+
+`;
+}
+
+export const loader = async () => {
+  // Fetch the markdown content server-side
+  const content = await fetchMarkdownFormSomewhere();
+  return { content };
+};
+
+export default function Index() {
+  // Access the markdown in your app
+  const { content } = useLoaderData<{ content: string }>();
+  // Render the markdown anywhere in your app (it also works server-side!)
+  return <MarkdownContainer source={content} />;
```

Comment:
Wouldn’t it be better to transform MD to HTML in the loader and return html directly? That way you don’t even need to ship MD parsing code to the browser

Reply:
Good point! Would that also work with custom React components?

This implementation supports both server-side and client-side markdown strings. E.g. you can also use it to implement a preview for a textarea with markdown support. What do you think? I can definitely refactor this to render the HTML server side!